### PR TITLE
allow custom input rendering by exposing a render function for it (fixed example)

### PR DESCRIPTION
**Pull request by angelo-v**
This is based on #16 but I fixed the example and the PropType

### CODE REVIEW

File: demo/src/components/reducers/app.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          focusedSectionIndex: focusedSectionIndex,
+          focusedItemIndex: focusedItemIndex
+        }
+      };
+    }
+
+    default:
+      return state;
+  }
+}
```
Comment (moroshko) [inline demo/src/components/reducers/app.js]: This whole file should be removed, as it's not used anymore.

File: src/Autowhatever.js [hunk 2]
```diff
@@ -30,6 +30,10 @@ export default class Autowhatever extends Component {
     shouldRenderSection: PropTypes.func,   // This function gets a section and returns whether it should be rendered, or not.
     renderSectionTitle: PropTypes.func,    // This function gets a section and renders its title.
     getSectionItems: PropTypes.func,       // This function gets a section and returns its items, which will be passed into `renderItem` for rendering.
+    inputElement: PropTypes.oneOfType([    // This element is used to render the input
+      PropTypes.func,
+      PropTypes.string
+    ]),
```
Comment (moroshko) [inline src/Autowhatever.js]: I believe we should allow `inputElement` to be a Component (that extends `React.Component`) as well.

Comment (angelo-v) [inline src/Autowhatever.js]: This is actually possible. I adjusted Example 10 to clarify this

Comment (novascreen) [inline src/Autowhatever.js]: Unfortunately `PropTypes.node` doesn't include functional stateless components, but `PropTypes.node` + `PropTypes.func` should cover it (that will cover string as well).

```js
// Anything that can be rendered: numbers, strings, elements or an array
// (or fragment) containing these types.
optionalNode: React.PropTypes.node,
```

Or is that too lenient? Would `func` + `element` work or doesn't that work for `'input'`?

Thanks for taking over @angelo-v, i was just gonna make the changes, but i guess you got it.

Comment (angelo-v) [inline src/Autowhatever.js]: I tried PropTypes.node but it did not work neither for Components, nor stateless functions nor strings. So I ended up with func + string which covers all cases.

**Comment by moroshko** [comment]
`inputElement` is available in [v5.4.0](https://github.com/moroshko/react-autowhatever/releases/tag/v5.4.0) now.

**Pull request merged**